packages/kbn-dev-utils/src/ci_stats_reporter/ci_stats_reporter.ts

```diff
@@ -90,19 +91,33 @@ export class CiStatsReporter {
       this.log.debug(e.message);
     }
 
+    try {
+      const { stdout } = await execa('git', ['branch', '--show-current']);
+      branch = stdout;
+    } catch (e) {
+      this.log.debug(e.message);
+    }
+
+    const memUsage = process.memoryUsage();
     const isElasticCommitter = email && email.endsWith('@elastic.co') ? true : false;
 
     const defaultMetadata = {
+      kibanaUuid,
+      isElasticCommitter,
       committerHash: email
         ? crypto.createHash('sha256').update(email).digest('hex').substring(0, 20)
         : undefined,
+      email: isElasticCommitter ? email : undefined,
+      branch: isElasticCommitter ? branch : undefined,
       cpuCount: Os.cpus()?.length,
       cpuModel: Os.cpus()[0]?.model,
       cpuSpeed: Os.cpus()[0]?.speed,
-      email: isElasticCommitter ? email : undefined,
       freeMem: Os.freemem(),
-      isElasticCommitter,
-      kibanaUuid,
+      memoryUsageRss: memUsage.rss,
```

Comment:
There's some higher level API's that look interesting here.  I don't think we should use them but could be a source of inspiration.

https://nodejs.org/api/report.html

Reply:
(not requesting any changes here)